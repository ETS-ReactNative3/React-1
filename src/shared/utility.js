import React, { Component } from 'react';

const STATUS = {
    HOVERED: 'hovered',
    NORMAL: 'normal',
};

export function handleModifyAnswerUpvotes(Answers, answerId, increment) {
    return Answers.map(answer => {
        if (answer.answerId != answerId)
            return answer;
        else {
            return { ...answer, upvotes: answer.upvotes + increment }
        }
    })
}

export const AddTwoNumbers = (a, b) => {
    return a + b;
}

export class CheckboxWithLabel extends Component {

    state = {
        isChecked: false,
    };

    onChange = () => {
        this.setState({ isChecked: !this.state.isChecked });
    };

    render() {
        return (
            <label>
                <input
                    type="checkbox"
                    checked={this.state.isChecked}
                    onChange={this.onChange}
                />
                {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
            </label>
        );
    }
}

export class Link extends Component {

    constructor() {
        super();

        this.state = {
            class: STATUS.NORMAL,
        };
    }
    _onMouseEnter = () => {
        this.setState({ class: STATUS.HOVERED });
    };

    _onMouseLeave = () => {
        this.setState({ class: STATUS.NORMAL });
    };

    render() {
        return (
            <a
                className={this.state.class}
                href={this.props.page || '#'}
                onMouseEnter={this._onMouseEnter}
                onMouseLeave={this._onMouseLeave}
            >
                {this.props.children}
            </a>
        );
    }
}

