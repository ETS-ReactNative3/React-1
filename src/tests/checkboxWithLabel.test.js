import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CheckboxWithLabel } from '../shared/utility'

it('check if checkbox rendered or not', () => {
    // Render a checkbox with label in the document
    const wrapper = shallow(<CheckboxWithLabel labelOn="On" labelOff="Off" />);

    expect(wrapper).toMatchSnapshot();
});

it('CheckboxWithLabel changes the text after click', () => {
    // Render a checkbox with label in the document
    const checkbox = shallow(<CheckboxWithLabel labelOn="On" labelOff="Off" />);

    expect(checkbox.text()).toEqual('Off');

    checkbox.find('input').simulate('change');

    expect(checkbox.text()).toEqual('On');
});