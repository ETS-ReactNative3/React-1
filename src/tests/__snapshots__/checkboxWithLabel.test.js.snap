// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if checkbox rendered or not 1`] = `ShallowWrapper {}`;
